lib/jit_preloader/active_record/associations/preloader/ar5_association.rb

```diff
@@ -53,7 +44,7 @@ def associate_records_to_owner(owner, records)
         # We don't want to duplicate them, but we also want to preserve
         # the original copy so that we don't blow away in-memory changes.
         new_records = association.target.any? ? records - association.target : records
-        association.target.concat(new_records)
+        association.target = association.target.concat(new_records)
```

Comment:
I'm confused by this. Shouldn't `association.target.concat` modify `association.target` in place?

Reply:
I'll make it more explicit, but modifying the `target` array doesn't mark the association as being loaded which results in it going back to the database.